src/block_parser.js

```diff
@@ -602,8 +593,19 @@ module.exports = function (args) {
       if (err) {
         return cb(err)
       } else {
-        return cb(err, '{ "txid": "' +  res + '" }')
+        cb(err, '{ "txid": "' +  res + '" }')
```

Comment:
Why wouldn't we call `cb` *after* parsing the transaction?
Current situation still leaves us with the chance of a race condition, and to get such sequence of events:
1. client calls `transmit`
2. client gets response (`cb` was called), and immediately calls for `getTransaction`
3. full-node responds with a transaction with *no assets*.
4. finally, full-node calls `decodeTransaction` and `addColoredIOs`.

It might have a slight performance impact, but I think it's negligible when doing that for a single transaction - and for the lease it would be good to measure the overhead.